dotcom-rendering/src/lib/branding.ts

```diff
@@ -8,3 +24,170 @@ export const pickBrandingForEdition = (
 	editionBrandings.find(
 		({ edition, branding }) => edition.id === editionId && branding,
 	)?.branding;
+
+/**
+ * Retrieve the branding object from each of the cards in an array of cards, for a given edition
+ *
+ * @returns `undefined` if AT LEAST ONE of the cards is missing branding,
+ * otherwise returns a non-empty array of branding
+ */
+const getBrandingFromCards = (
+	cards: CardWithBranding[],
+	editionId: EditionId,
+): NonEmptyArray<Branding> | undefined => {
+	const brandings: Branding[] = [];
+	for (const card of cards) {
+		const branding = pickBrandingForEdition(
+			card.properties.editionBrandings,
+			editionId,
+		);
+		// If a single card is missing branding then we bail out immediately
+		if (!branding) {
+			return undefined;
+		}
+		brandings.push(branding);
+	}
+	if (!isNonEmptyArray(brandings)) {
+		return undefined;
+	}
+	return brandings;
+};
+
+const getBrandingType = ([
+	firstBranding,
+	...restBranding
+]: NonEmptyArray<Branding>): BrandingKind | undefined => {
+	const name = firstBranding.brandingType?.name;
+
+	if (!name) {
+		return undefined;
+	}
+
+	const allNamesMatch = restBranding.every(
+		({ brandingType }) => brandingType?.name === name,
+	);
+
+	if (!allNamesMatch) {
+		return undefined;
+	}
+
+	return name;
+};
+
+/**
+ * Check each branding has the same sponsor name
+ */
+const everyCardHasSameSponsor = ([
```

Comment:
is this going to be true if all the cards have branding undefined? I think it might

Reply:
`getBrandingFromCards` checks that every card has a branding (for the given edition) before we call `everyCardHasSameSponsor`, so this function is only called once we're sure all the cards have branding.

I'll add a test for this though.

Reply:
Ah very good, I had missed that. Thanks!